src/components/ExpandableText/ExpandableText.tsx

```diff
@@ -0,0 +1,82 @@
+import * as React from 'react';
+import ExpandIcon from '@material-design-icons/svg/round/expand_more.svg';
+import { bem } from '../../utils';
+import styles from './ExpandableText.scss';
+import { Button } from '../Buttons';
+
+export interface Props extends React.HTMLAttributes<HTMLDivElement> {
+    /** The content of the block */
+    children: React.ReactNode;
+    /** The maximum number of lines to be shown when displaying the text in collapsed mode  */
+    threshold?: number;
+    /** Label for the button when the element is collapsed */
+    showMoreButtonLabel: string;
+    /** Label for the button when the element is expanded */
+    showLessButtonLabel: string;
+}
+
+const { block, elem } = bem('ExpandableText', styles);
+
+export const ExpandableText = React.forwardRef<HTMLDivElement, Props>((props, ref) => {
+    const { children, threshold = 5, showMoreButtonLabel, showLessButtonLabel, ...rest } = props;
+    const contentRef = React.createRef<HTMLDivElement>();
+    const [isExpandable, setIsExpandable] = React.useState(false);
+    const [isCollapsed, setIsCollapsed] = React.useState(true);
+
+    const collapsedHeight = `calc(var(--line-height-normal) * ${threshold})`;
+
+    // After first render determine if the text is longer or not then the maximum allowed
+    React.useLayoutEffect(() => {
+        if (!isExpandable && contentRef.current) {
+            contentRef.current.style.maxHeight = collapsedHeight;
+
+            if (contentRef.current.scrollHeight > contentRef.current.clientHeight) {
+                setIsExpandable(true);
+                contentRef.current.style.height = collapsedHeight;
+                contentRef.current.style.maxHeight = 'initial';
+            }
+        }
+    }, []); // eslint-disable-line react-hooks/exhaustive-deps
+
+    const collapseSection = () => {
+        if (!contentRef.current) {
+            return;
+        }
+
+        contentRef.current.style.height = collapsedHeight;
+    };
+
+    const expandSection = () => {
+        if (!contentRef.current) {
+            return;
+        }
+
+        const sectionHeight = contentRef.current.scrollHeight;
+        contentRef.current.style.height = `${sectionHeight}px`;
+    };
+
+    const handleToggleClick = () => {
+        if (isCollapsed) {
+            expandSection();
+        } else {
+            collapseSection();
+        }
+        setIsCollapsed((prevIsCollapsed) => !prevIsCollapsed);
```

Comment:
I'd move run function `setIsCollapsed` to the methods `expandSection` and `collapseSection`, otherwise running those function without `setIsCollapsed` will be break UI.

Reply:
Sorry, I'm not sure I understand what do you mean. Can you elaborate?

Reply:
Imagine I need to run functions `expandSection` and `collapseSection` also from another place. In this case, the height will be changed, but state `isCollapsed` will be the same. So the idea to add `setIsCollapsed` inside those methods:


```
    const expandSection = () => {
        if (!contentRef.current) {
            return;
        }

        const sectionHeight = contentRef.current.scrollHeight;
        contentRef.current.style.height = `${sectionHeight}px`;
        setIsCollapsed(true);
    };
```

Reply:
but it is not possible to call these functions from somewhere else. 🤔 They aren't exposed, and that was not a requirement either. How would you call these from outside?

Reply:
I understand they aren't exposed, but the concern is about maintaining consistency within the component. By integrating setIsCollapsed within these methods, we ensure that the component's state remains consistent regardless of how these methods are invoked.